### PR TITLE
feat(docker): fetch docker pulls limits

### PR DESCRIPTION
**Pull request by chiptus**
related to https://github.com/portainer/portainer/pull/4666

### CODE REVIEW

File: http/handler/dockerhub/dockerhub_status.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, errors.New("failed fetching dockerhub limits")
+	}
+
+	rateLimit, err := parseRateLimitHeader(resp.Header, "RateLimit-Limit")
+	rateLimitRemaining, err := parseRateLimitHeader(resp.Header, "RateLimit-Remaining")
```
Comment (sbusso) [inline http/handler/dockerhub/dockerhub_status.go:119]: As for portainer we would need to handle those errors.

File: http/handler/dockerhub/dockerhub_status.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+	return response.JSON(w, resp)
+}
+
+func getDockerHubToken(httpClient *http.Client, dockerhub *dockerhubStatusPayload) (string, error) {
+	type dockerhubTokenResponse struct {
+		Token string `json:"token"`
+	}
+
+	requestURL := "https://auth.docker.io/token?service=registry.docker.io&scope=repository:ratelimitpreview/test:pull"
```
Comment (sbusso) [inline http/handler/dockerhub/dockerhub_status.go:67]: as for portainer we should use some constant

**Review by sbusso — CHANGES_REQUESTED** [review]
small change requests

File: http/handler/dockerhub/dockerhub_status.go (hunk 2 above)
Comment (chiptus) [reply http/handler/dockerhub/dockerhub_status.go]: this code should be in a library :(

**Review by sbusso — APPROVED** [review]

**Review by sbusso — APPROVED** [review]

**Pull request merged**